src/pint/models/wavex.py

```diff
@@ -0,0 +1,372 @@
+"""Delays expressed as a sum of sinusoids."""
+import astropy.units as u
+import numpy as np
+from loguru import logger as log
+from warnings import warn
+
+from pint.models.parameter import MJDParameter, floatParameter, prefixParameter
+from pint.models.timing_model import DelayComponent, MissingParameter
+
+
+class WaveX(DelayComponent):
+    """Implementation of the wave model as a delay correction
+
+    Delays are expressed as a sum of sinusoids.
+
+    Used for decomposition of timing noise into a series of sine/cosine components with the amplitudes as fitted parameters.
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.wavex.WaveX
+
+    This is an extension of the L13 method described in Lentati et al., 2013
```

Comment:
You might want more info here on how this compares to the `Wave` model, and how to translate back & forth.

Reply:
Maybe even some code snippets.

Reply:
also specify more info about which Lentati paper this is

Reply:
Added more information to the docstring, including DOI for the Lentati paper